Feature/physx 4.0

webhaikal opened the pull request:
This PR solves #252. Update PhysX from 3.3 to 4.0, CMake is updated to reflect this but as I'm not familiar with CMake, it might have some errors.

Following on above, you would need to add another value to C++/C->Preprocessor or equivalent the following item:
`PX_PHYSX_CHARACTER_STATIC_LIB`
to avoid linking errors.

I've tested it using the bsfExamples but couldn't make the ball spawn, so I can't test rigidbodies. So you can treat this as WIP. You may also revert commit webhaikal/bsf@318fe2814a5cc298871302c4f590bebde10a5a78

BearishSun committed:
Feature: BS_PARAMS can be used on method parameters to export a Vector as a variable parameter entry in the script code

BearishSun committed:
Refactor: Major refactor of `IReflectable`
 - `mRTTIType` field was removed. It was being used for temporary data storage during serialization/deserialization, but it was intrusive and part of every `IReflectable` object
 - Temporary data can be now stored directly on `RTTIType` which is now guaranteed to be unique per `IReflectable` instance. This ultimately yields much cleaner code, less dynamic allocations and most importantly `IReflectable` is now a pure interface with no data types.

BearishSun committed:
Build: Forgot to push a file with previous commit
 - Also some cleanup

BearishSun committed:
WIP: Particle system
 - Game object handle for the particle system component

BearishSun committed:
Feature: Added a Bitfield container

hachmeister committed:
Bugfix: Generate mPointerDelta and event.delta before event is fired (#184)

BearishSun committed:
WIP: Particle system
 - Initial work on GPU particles

BearishSun committed:
Build: Fixing a compilation error with GCC/Clang introduced with previous commit

BearishSun committed:
Refactor: More unified GpuBuffer handling across different APIs
 - GpuBuffer can now also be created with an external buffer, acting as a buffer view
 - Vertex/index buffer types can now have a load store flag (not fully exposed)
 - Few other minor tweaks and fixes

BearishSun committed:
Feature: Vertex/Index buffers can now be bound as generic load-store buffers
 - Also cleaned up various buffer related render API specific code and moved a lot of it to base classes

BearishSun committed:
Bugfix: Fixing errors reported by Vulkan validation layers for 1.1.77
 - External subpass dependency access mask should be 0
 - Buffer views shouldn't get destroyed until they are done being used on the command buffer

BearishSun committed:
Bugfix: (Vulkan) Issue a memory barrier if vertex/index buffer has been written to by a shader

BearishSun committed:
Refactor: Consistent naming for HardwareBuffer implementations across render backends

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
WIP: Particle system
 - More work on GPU particles

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
Bugfix: Buffers with a 16-bit unsigned integer format now report proper element size

BearishSun committed:
WIP: Particle system
 - Modified particle indices to use x/y offset to make the particle rendering code compatible with both CPU and GPU systems

BearishSun committed:
WIP: Particle system
 - GPU particle simulation hooked up to the CPU frontend and (for the most part) to the renderer

BearishSun committed:
Build: Fixing GCC compile error caused by previous commit

BearishSun committed:
WIP: Particle system
 - Adding helper classes for evaluating curves on the GPU

BearishSun committed:
Feature: Added a NO_RREF attribute to allow resource references to be exported as raw values instead of RRefs in script code

BearishSun committed:
Bugfix: Properly apply binary diffs

BearishSun committed:
WIP: Particle system
 - Add a shader for calculating particle bounds on the GPU

BearishSun committed:
Bugfix: MeshData iterators now properly report when they reach the end

BearishSun committed:
Build: Fixing broken macOS build

BearishSun committed:
Build: Fixing build issues on Linux

BearishSun committed:
Bugfix: Fixing a crash due to Vulkan framebuffer getting a delayed update on Linux

BearishSun committed:
Bugfix: When creating a Vulkan swap chain make sure to account for that fact that swap chain might not be the one originally requested

BearishSun committed:
Bugfix: Destoying Vulkan buffer views will now properly wait until the buffer is done being used on the GPU

BearishSun committed:
Build: Update Appveyor FTP user info

BearishSun committed:
WIP: Particle system
 - Particle rendering shader now accepts outputs from the GPU particle simulation

BearishSun committed:
WIP: Particle system
 - GPU simulation & rendering running (nothing gets simulated yet though)

BearishSun committed:
WIP: Particle system
 - Bug fixes for the texture row allocator
 - Unit tests for the texture row allocator and distribution -> lookup table conversion

BearishSun committed:
WIP: Particle system
 - Vector field GPU simulation functional

BearishSun committed:
Build: Fixing GCC compile issues

BearishSun committed:
Refactor: Added helpers for easier core object syncing with significantly less boilerplate

BearishSun committed:
Refactor: Added helper types to use in types referencing both variants of CoreObjects - reduces boilerplate

BearishSun committed:
Build: Fixing FMOD compile errors

BearishSun committed:
Feature: GUI elements can now report screen space bounds

BearishSun committed:
WIP: Particle system
 - Size and rotation properties are now handled by the GPU simulation
 - Added helper class for uploading curves for use on the GPU

BearishSun committed:
Bugfix: Fix divide by zero issue in certain audio system components

BearishSun committed:
Bugfix: Fix a memory leak in FMOD audio plugin

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
WIP: Particle system
 - Curve evaluation on the GPU hooked up
 - Added color over lifetime & size over lifetime modifiers to GPU simulation

BearishSun committed:
WIP: Particle system
 - Texture animation now supported when using GPU simulated particles

BearishSun committed:
WIP: Particle system
 - Added depth buffer collisions for GPU simulated particles

BearishSun committed:
WIP: Particle system
 - Added constant acceleration and drag forces to the GPU simulation
 - Added random offset property to the emitter
 - Depth collisions now use accurate particle size
 - Particles spawned for GPU simulation now correctly ignore maxParticles property in all cases
 - GPU particle simulation now correctly only triggers for a single view per frame

BearishSun committed:
WIP: Particle system
 - GPU simulated particles can now be sorted

BearishSun committed:
Refactor: GUIElements can now trigger content updated on other GUI elements, during a content update

BearishSun committed:
Feature: Added an utility to calculate a range of a set of animation curves

BearishSun committed:
Feature: Added script bindings for ParticleDistribution and Random types

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
WIP: Particle system
 - User can now provide custom bounds to the particle system

BearishSun committed:
WIP: Particle system
 - Added 3D mesh particle rendering

BearishSun committed:
WIP: Particle system
 - Particles now support lighting along both forward and deferred rendering paths

Milerius committed:
Feature: Added fixedUpdate function for fixed update from the user side. (#197)

Feature: Added fixedUpdate function for fixed update from the user side.

Milerius committed:
Fixing [#198]: update json.hpp to repair the project compilation on clang 7.0 (#199)

BearishSun committed:
WIP: Particle system
 - Particle materials now support lighting

BearishSun committed:
Build: Fixing macOS build

BearishSun committed:
Bugfix: Gaussian blur filter now properly samples along the provided filter radius

BearishSun committed:
Feature: Added a Bloom effect
 - And a few fixes/extensions for the downsample & blur shaders

BearishSun committed:
Refactor: Fixing some naming issues

BearishSun committed:
Bugfix: Reverting back DX11 texture array binding 'fix' that broke texture arrays
 - Fixing the warning in ClearLoadStore shader in an alternative way instead

BearishSun committed:
Refactor: Slightly improved CPU profiler coverage

BearishSun committed:
Bugfix: OpenAudio undefined symbol issue on Linux
 - Also cleaning up some warnings

BearishSun committed:
Bugfix: Properly render opaque particles using the deferred path

BearishSun committed:
Refactor: Core sync RTTI refactored so it can be called recursively

BearishSun committed:
Refactor: Eliminating redundant dynamic allocation from IReflectable::_getRTTIfromTypeId

ondrowan committed:
Rename BSPlatform to BSFPlatform (#201)

It clashed with class from MacOS's BaseBoard framework with the same name.

BearishSun committed:
Refactor: Variety of core sync methods moved to the new approach

BearishSun committed:
Refactor: Better design and performance for serializer
 - Binary serializer now has a dedicated decode() method that doesn't
 use intermediate serialization, which is expensive due to all the
 dynamic memory allocations
 - Intermediate serialization has been moved together with
 SerializedObject, instead of being part of BinarySerializer
 - Added a dedicated encode() method to intermediate serialization, that
 doesn't use binary serialization as an intermediary

DNKpp committed:
BsCCapsuleCollider calculate local rotation from normal on construction (#202)

* BsCCapsuleCollider calculate local rotation from normal on construction

Signed-off-by: DNKpp <DNKpp2011@gmail.com>

* add normalize in bs::CCapsuleCollider::setNormal

nemerle committed:
Fix `throw` being used by simdpp's aligned_allocator (#205)

closes #204

BearishSun committed:
Bugfix: Parsing C# generic type name will sometimes fail, don't crash
 - Also added a helper method for retrieving C# generic parameters

BearishSun committed:
Feature: Particle system can now spawn particles in bursts

BearishSun committed:
Bugfix: Emission bursts now don't skip cycles with small intervals and handle loops properly

BearishSun committed:
Build: Fixing a compile error introduced with last commit

BearishSun committed:
Feature: New non-randomized particle emission modes
 - Sequential emission modes added to circle, line, cone and mesh emitters

BearishSun committed:
Bugfix: Changing component state would result in an infinite loop

BearishSun committed:
Bugfix: Force resource loading to be synchronous if async is not supported by resource type

BearishSun committed:
Refactor: Game object serialization refactored so it can be done safely from multiple threads

BearishSun committed:
Feature: Emissive materials

remoe committed:
fix #208 build issue (#209)

remoe committed:
copyToMemory is not needed (#211)

Fix for #210

guestnone committed:
Build: Add Windows Resource files and extra config values (#212)

This commit adds description for compiled binaries on Windows platforms
for easier version identification. It also adds additional configuration
values that can be set in main CMakeLists.txt file:
  * BS_VERSION_PATCH for very minor bug-fixing versions of the framework.
  * BS_CURRENT_RELEASE_YEAR which is self explanatory.

Plus the BS_VERSION_STRING macro for providing the version of the framework as a string.

BearishSun committed:
Build: Fixing FMOD audio module build issue

BearishSun committed:
Feature: Added size/color/rotation/force over lifetime particle evolvers

paolopaoletto committed:
Refactor: operator methods in Color class (#214)

BearishSun committed:
Feature: Materials variations can now be toggled by the user, per material

BearishSun committed:
Feature: Soft particles

paolopaoletto committed:
Custom SmallVector implementation

BearishSun committed:
Feature: BSL importer now allows selection of render backends to generate shader code for

BearishSun committed:
Build: Potentially fixing macOS build

BearishSun committed:
Build: Built-in asset import tool added
 - Built-in assets are no longer processed on application start-up, sometimes causing extreme start-up times (esp. in debug mode)
 - Automated asset dependency versioning & uploaded added
 - Shader binaries are no longer part of Git as they have started being too large

BearishSun committed:
Build: More fixes for the macOS build

BearishSun committed:
Feature: Integrated SmallVector container into the framework

BearishSun committed:
WIP: Decals
 - Initial skeleton

BearishSun committed:
Refactor: Import tool will now retrieve UUIDs for all built-in resources from the data list JSON

BearishSun committed:
Refactor: Import tool will now retrieve UUIDs for all built-in resources from the data list JSON

BearishSun committed:
WIP: Decals

BearishSun committed:
Bugfix: Import tool will now ignore asset groups not part of data list JSON

BearishSun committed:
WIP: Decals
 - Improved BSL preprocessor

BearishSun committed:
Build: Include the resource manifest in the compiled data archive

BearishSun committed:
Bugfix: Import tool now properly processes sprite icons

BearishSun committed:
Refactor: Sequential mesh emission modified so it never skips a vertex

remoe committed:
VulkanImage getSubresource boundary check (#219)

BearishSun committed:
WIP: Decals

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
Bugfix: Properly set up animated sprite textures on the material

BearishSun committed:
WIP: Decals

BearishSun committed:
Bugfix: Fix import for 64-bit per-channel non-floating-point textures

BearishSun committed:
Docs: Some work on the particle system documentation

BearishSun committed:
WIP: Particle system
 - Exposed parts of the particle system to scripting

remoe committed:
remove unused code (#221)

BearishSun committed:
WIP: Particle system
 - More of the system exposed to scripting

BearishSun committed:
WIP: Decals
 - Decals now properly render when viewer is inside their bounds
 - Added proper MSAA handling for decals
 - Some fixes to the render queue so it doesn't apply unnecessary passes

paolopaoletto committed:
Feature: Added a DynArray container

BearishSun committed:
Feature: Deferred decals now supported
 - Finished current decal implementation with support for masking on which surfaces are decals projected

BearishSun committed:
Docs: Added documentation about various particle emitter shapes

BearishSun committed:
Docs: Added documentation for particle evolvers

BearishSun committed:
Bugfix: Fix crash when bone is being removed from an animation

BearishSun committed:
Bugfix: Always round buffer size returned from glslang to 16 bytes

BearishSun committed:
Bugfix: Properly deduce sampler border color in Vulkan

BearishSun committed:
Refactor: Particle system emitters & evolvers are now contained in normal vectors, to make scripting simpler

BearishSun committed:
Feature: Particle collision planes can now be automatically fetched from linked scene objects

BearishSun committed:
Refactor: Skinned mesh particle emitter can now accept a component CRenderable as well as the actor
 - Also added a general-purpose container for similar cases

xytovl committed:
Allow deactivation of most bundled libraries (#225)

BearishSun committed:
Docs: Added documentation for particle system settings

Meumeu committed:
Cleanup the build system

Put generated files in the build directory
Use target properties instead of CMAKE_CXX_FLAGS and CMAKE_LINKER_FLAGS
Install a cmake module
Install bundled dependencies in a subdirectory in linux to not pollute the system libs
Don't install .version / .reqversion in the data directory

xytovl committed:
Fix increment in cubemap faces loading (#226)

increment was performed before loading, causing an off-by-one error on
faces

BearishSun committed:
Bugfix: (Vulkan) Always bind a texture array using the texture array view

BearishSun committed:
Bugfix: (Vulkan) Fix an issue where image subresource layouts would fail to be set properly in some cases

BearishSun committed:
Bugfix: Generate filtered reflection for the lowest mipmap level

BearishSun committed:
Bugfix: Fix a crash when unregistering a bone from an animation

Meumeu committed:
This should fix compilation errors with Windows and OSX

Meumeu committed:
Remove /EHsc from CMAKE_CXX_FLAGS for MSVC

This is necessary to disable exceptions as /EHsc is automatically
added by cmake and we cannot remove flags with target properties.

Commit:
Build: Make run to link with the static version for LibUUID

Commit:
Build: Fixing a linker error on macOS, as well as various warnings

BearishSun committed:
Docs: Added documentation for GPU simulated particles
 - Also cleaned up various parts of the documentation

BearishSun committed:
Build: Updated minimum CMake version to 3.12.4

BearishSun committed:
Build: Lessen the CMake version back to 3.12.0 for now, so the CI doesn't fail

BearishSun committed:
Build: Update Travis CI CMake version

BearishSun committed:
Build: Grab latest CMake on mac OS CI

BearishSun committed:
Docs: Documentation for decals
 - Also various images for the particle system docs

Meumeu committed:
Fix linker flags for MSVC

Meumeu committed:
Use relative RPATH

BearishSun committed:
Feature: Particle system continuous (sub-frame) emission

BearishSun committed:
Bugfix: Fixing script export for particle evolvers

Meumeu committed:
Copy dependencies DLL in the build directory for the Windows version

BearishSun committed:
Bugfix: Properly export some particle system methods to script

BearishSun committed:
Refactor: Added parameterless constructor methods for particle evolvers & emitter shapes, mainly for script purposes

BearishSun committed:
Feature: Added a dynamic cast `rtti_cast` method

BearishSun committed:
Build: Fix Linux build

BearishSun committed:
Docs: Updated build guide to match the new process

BearishSun committed:
Docs: Particle surface shaders and other doc additions

paolopaoletto committed:
Feature: Class to contain Complex numbers

BearishSun committed:
Build: Fix Vulkan build

Commit:
Bugfix: (Vulkan) OUT_OF_DATE error will now trigger a rebuild of the swap chain, instead of relying only on the window manager for notifying us

Commit:
Bugfix: (Vulkan) Fix integrated GPU enumeration

BearishSun committed:
Build: Fix nightly build naming

BearishSun committed:
Docs: Added particles to the feature list

Commit:
Bugfix: (Vulkan) Fix unnecessary use of TU_LOADSTORE flag on multisample textures, as they are unsupported on Intel IGPUs

BearishSun committed:
Bugfix: (Vulkan) Fix layout transitions not being properly remember after queue submit
 - This happened if render target was set and its render pass never got executed before the submit (no draw calls)
 - Or if initial layout was undefined (the case with many attachment types) the system would just not set the render pass final layout

BearishSun committed:
Build: Documentation images are now in their own asset package instead of the git repo

Commit:
Bugfix: (Vulkan) Fix indirect diffuse calculations not preserving the depth buffer
 - Was causing issues on Intel IGPU

Commit:
Feature: Added sRGB <-> linear color space conversion methods

Commit:
Bugfix: Fixing application icon on Linux

Commit:
Bugfix: (Vulkan) Respect load flags when clearing the render target

Commit:
Bugfix: SSRTrace shader now properly preserves depth-stencil and loads it as read-only

paolopaoletto committed:
Feature: MinHeap class for geometric algorithm.

The class can be used as a priority queue in some geometric algorithm. For example: Polyline Reduction.
The difference with the standard implementation is mostly in the performance. Using the standard implementation in
Polyline Reduction there is no direct access to the vertices modified, so it must be made a searching action, remove action and after
update the weight of the nodes, they must be reinserted. This class avoid the removing and the reinserting.

BearishSun committed:
Bugfix: Fix a crash when destroying a particle system

BearishSun committed:
Bugfix: Make sure vector field precompiled asset can be properly loaded as a resource

BearishSun committed:
Refactor: Asset import and upload are now build targets instead of ran at configure time

BearishSun committed:
Build: Moving some build options so they are inherited by parent projects

BearishSun committed:
Refactor: Particle orientation plane really needs to be only a direction rather than a plane

BearishSun committed:
Refactor: AnimationUtility refactored to support split & combine functionality on arbitrary animation curve types

Commit:
Build: Fixing RPATH and imported library install for shared libraries

Commit:
Build: Resolving some potential issues with Vulkan memory barriers

Commit:
Bugfix: Resolving a driver hang on Intel GPUs

Commit:
Build: Fixing a compile error with Clang 7, and some general compile issues with SmallVector

BearishSun committed:
Bugfix: Updated Vulkan device enumeration when choosing a non-0 device

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
Refactor: Big refactor of Vulkan image layout transitions, memory and execution barriers to make them more correct, easier to follow and more performant

BearishSun committed:
Refactor: Distributions now store the constant values within the curve/gradient so the same getter methods can be used regardless of the actual distribution type
 - Also fixed an issue in ColorGradient that returned invalid colors when sampling a time before the first keyframe

BearishSun committed:
Bugfix: Vector deserialization doesn't first clear the vector

BearishSun committed:
Bugfix: GPU particle simulation now works properly on OpenGL

BearishSun committed:
Bugfix: (Vulkan) Modifying an index/vertex buffer after it has been bound will now properly use the modified buffer on subsequent draw calls

BearishSun committed:
Bugfix: GPU particle tile clear now properly clears all particle textures

BearishSun committed:
Bugfix: Import async operations can now be waited on

BearishSun committed:
Bugfix: Particle depth buffer collisions now work properly on local-space particle systems

Commit:
Bugfix: (Vulkan) Fix render target load flags not being properly applied to depth/stencil buffer

Commit:
Bugfix: Workaround for Mesa driver bug when accessing row-major matrices in an UBO/SSBO

Commit:
Build: Missed a file in the previous commit

BearishSun committed:
WIP: Particle editor export
 - Added some attributes so that particle system settings are correctly handled by the editor inspector

paolopaoletto committed:
Feature: Romberg Integration and Gaussian Quadrature.

Commit:
Build: Fixing rpath on macOS

Commit:
Build: Hide OpenGL deprecation warnings on macOS
 - Also fail earlier when OpenGL context fails to be initialized

BearishSun committed:
Bugfix: Fixing UV seam on the default sphere mesh
 - Also working around XShaderCompiler codegen issue for GLSL on macOS

BearishSun committed:
Bugfix: Avoid blocking the main thread when submitting the core thread commands

Commit:
Bugfix: Fix OpenGL not rendering in macOS Mojave

BearishSun committed:
Build: Moving CI deployment to S3

BearishSun committed:
Docs: Added missing information about particle system and decals to the renderer documentation

BearishSun committed:
Bugfix: Fixing particle system threading issue

BearishSun committed:
Bugfix: Fixing an issue where the task scheduler would use up all available threads in the pool

BearishSun committed:
Bugfix: Invalid task scheduler comparator

BearishSun committed:
Bugfix: Default GUISkin should be loaded from GUISkin.json.asset, not GUISkin.asset

BearishSun committed:
Bugfix: Have the renderer resize the internal render targets when the output render target resizes

BearishSun committed:
Bugfix: Fix emissive materials not rendering properly when using the non-compute rendering path

paolopaoletto committed:
Feature: PI and Square Root constant variations.

BearishSun committed:
Build: Automated documentation generation and upload

guestnone committed:
Feature: Methods for getting current date/time as text.
- Add the application startup date and a method for getting it as a text.

guestnone committed:
Refactor: Improvements to the logging system.
- Add the local date of receiving the log message.
- Improve the look of the HTML log file and made it
  so it detects if the framework was compiled standalone
  or for Banshee Engine.
- Debug channels now gets outputted to debugger/console.

BearishSun committed:
Build: Fixing CI deployment, documentation generation should now include images

BearishSun committed:
Docs: Added version selector to documentation

paolopaoletto committed:
Refactor: Modernized constructors for various audio system classes

BearishSun committed:
Docs: Update supported Vulkan SDK version

BearishSun committed:
Bugfix: Multiple decal related fixes
 - Layer mask texture is not longer integer as it's not supported for multisampling on macOS
 - Decal vertex shader no longer requires UV, which was failing on Vulkan as UV is not provided by the vertex buffer
 - Normal tolerance comparison now works properly on OpenGL

BearishSun committed:
Bugfix: Material deserialization issues

BearishSun committed:
Refactor: Better error messages when creating invalid index or vertex buffers

BearishSun committed:
Docs: Improved build guide and cleaned up some Doxygen warnings

BearishSun committed:
Bugfix: Fix material parameters not maintaining order during serialization

paolopaoletto committed:
Refactor: Modernized constructor for FLACDecoder and OggVorbis system class. (#247)

Refactor: Modernized constructor for FLACDecoder and OggVorbis system class

BearishSun committed:
Bugfix: Improved physics simulation stability during perf. spikes

BearishSun committed:
Build: Fixing macOS compile issues caused by previous commit

Commit:
Bugfix: Fix animation manager not returning the latest available animation results

ataulien committed:
Feature: Add Distance Culling

BearishSun committed:
Build: Fix missing least/most significant bit intrinsics for 32-bit platforms

paolopaoletto committed:
Refactor: Modernized constructors

BearishSun committed:
Refactor: RendererUtility::drawScreenRect now uses a ring buffer instead of relying on write-discard
 - This works around the issue on macOS which seems to have a driver bug regarding write-discard

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

BearishSun committed:
Docs: Preparation for v1.1 release

BearishSun committed:
Merge branch 'release-v1.1.0'

paolopaoletto committed:
Refactor: Modernized constructor for Localization, Manager and other system classes.

paolopaoletto committed:
Docs: Adding Copyright and comments to files where they were missing.

BearishSun committed:
Refactor: RenderTarget and its derived types are now reflectable, so they can be cast dynamically

BearishSun committed:
Refactor: Particle emitter shape is now returned as a shared_ptr, so it gets properly exported to script code
 - Also some documentation cleanup

BearishSun committed:
Docs: Updated CONTRIBUTORS.md

paolopaoletto committed:
Refactor: Modernized constructor for Components, RenderAPI and other system classes.

webhaikal committed:
Updated PhysX to 4.0

Changes made:
* PxConvexFlag::eINFLATE_CONVEX is removed in 4.0, suggest to find other way around
* Not sure line 235 on BsPhysX.cpp does, commented it since we're no longer using the old pair.contactStream
* PxSceneQueryHitType is renamed to PxQueryHitType

Signed-off-by: Haikal Izzuddin <haikalizz05@gmail.com>

webhaikal committed:
[DNM] Small changes to CMake to avoid downloading documentation images

Signed-off-by: Haikal Izzuddin <haikalizz05@gmail.com>

webhaikal committed:
Merge branch 'master' of https://github.com/GameFoundry/bsf into feature/physx-4.0

webhaikal committed:
Update for Physx 4.0

Signed-off-by: Haikal Izzuddin <haikalizz05@gmail.com>

BearishSun committed:
Refactor: Particle system tweaks required for preview in editor

webhaikal committed:
Merge pull request #1 from GameFoundry/master

Merge head fork

webhaikal committed:
Merge pull request #2 from webhaikal/master

Merge master

BearishSun commented:
Thank you. Since this requires me to compile PhysX4 on all platforms, it will take longer than usual to merge, until I get a moment to do so. Let me know when you feel everything is functional on Windows and I'll see about merging.

I believe the stuff in `CMakeFiles` folder was added in error? Could you remove it from the PR?
Why is the documentation download commented out? In any case, it shouldn't be part of the PR.
I notice the event callback is commented out. Is there a replacement mechanism for this?
What's the issue with `eINFLATE_CONVEX`?

paolopaoletto committed:
Refactor: Modernized destructor for Audio system classes. (#257)

webhaikal commented:
The CMakeFiles & Documentation part was added in by error. Will revert it back. However, you would need to update the FindPhysx CMake file to account for the new naming scheme.

Event Callback was commented out as I am not sure how to implement but will update it around today.

eINFLATE_CONVEX is removed from PhysX 4 as the new algorithm doesn't need to have it inflated.

BearishSun commented:
Sounds good.

webhaikal committed:
Merge remote-tracking branch 'origin_bsf/master' into feature/physx-4.0

webhaikal committed:
Update PhysXEventCallback to reflect Physx 4.0 changes

Signed-off-by: Haikal Izzuddin <haikalizz05@gmail.com>

webhaikal committed:
Revert commit webhaikal/bsf@318fe2814a5cc298871302c4f590bebde10a5a78

Signed-off-by: Haikal Izzuddin <haikalizz05@gmail.com>

webhaikal commented:
> Sounds good.

There you go, everything is done, reverted back the documentation comments and updated FindPhysX to reflect new naming scheme. Everything should be functional, just the extra preprocessor flag is needed for bsPhysx `PX_PHYSX_CHARACTER_STATIC_LIB`
